SQL-Solutions.txt

```diff
@@ -0,0 +1,2743 @@
++------+---------+--------+--------+------+
+| 2001 | Hoffman | London |    100 | 1001 |
+| 2006 | Clemens | London |    100 | 1001 |
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+-- 3) Find the largest order taken by each salesperson on each date. 
+
+Query - 
+
+SELECT 
+    o.odate,
+    o.onum,
+    MAX(o.amt) AS largest_order_in_amt,
+    o.snum,
+    s.sname
+FROM
+    orders AS o
+        INNER JOIN
+    salespeople AS s ON o.snum = s.snum
+GROUP BY s.snum , o.odate
+ORDER BY o.snum;
+
+Output - 
+
++------+------------+----------------------+------+---------+
+| onum | odate      | largest_order_in_amt | snum | sname   |
++------+------------+----------------------+------+---------+
+| 3003 | 1990-10-03 |               767.19 | 1001 | Peel    |
+| 3008 | 1990-10-05 |                 4723 | 1001 | Peel    |
+| 3011 | 1990-10-06 |              9891.88 | 1001 | Peel    |
+| 3005 | 1990-10-03 |              5160.45 | 1002 | Serres  |
+| 3007 | 1990-10-04 |                75.75 | 1002 | Serres  |
+| 3010 | 1990-10-06 |              1309.95 | 1002 | Serres  |
+| 3009 | 1990-10-04 |              1713.23 | 1003 | AxelRod |
+| 3002 | 1990-10-03 |               1900.1 | 1004 | Motika  |
+| 3001 | 1990-10-03 |              1098.16 | 1007 | Rifkin  |
++------+------------+----------------------+------+---------+
+
+-- 4) Arrange the Order table by descending customer number. 
+
+Query - 
+
+SELECT 
+    *
+FROM
+    orders
+ORDER BY cnum DESC;
+
+Output - 
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+-- 5) Find which salespeople currently have orders in the order table. 
+
+Query - 
+
+SELECT 
+    *
+FROM
+    salespeople
+WHERE
+    snum IN (SELECT 
+            snum
+        FROM
+            orders);
+
+Output - 
+
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
+| 1004 | Motika  | London    |   11 |
+| 1007 | Rifkin  | Barcelona |   15 |
++------+---------+-----------+------+
+
+-- 6) List names of all customers matched with the salespeople serving them. 
+
+Query -
+
+SELECT 
+    c.cname AS CUSTOMER_NAME, s.sname AS SERVING_BY
+FROM
+    customers AS c,
+    salespeople AS s
+WHERE
```

Comment:
Why you have used cross join? can't you use another Type Join ? reason for doing this?

Reply:
Improved all queries with proper joins